### PR TITLE
PR for ticket 1477

### PR DESCRIPTION
**Pull request by kreczko**
https://svnweb.cern.ch/trac/cactus/ticket/1477#ticket

 - added new class MonitoringSetting
 - added MonitoringSetting table to GateKeeper
 - added XML parsing for ```<state>```, ```<metric>``` and ```<mon-obj>```

Other fixes/changes:
 - JSON files: SystemCreator -> System
 - added unit tests for (Xml)GateKeeper
 - added ```monitoring::Status``` to string (and back) mapping
 - fixed logging names and colors (FATAL was missing, DEBUG1 and DEBUG2 were swapped)

### CODE REVIEW

File: core/include/swatch/core/AbstractMetric.hpp [hunk 1]
```diff
@@ -35,6 +36,19 @@ enum Status {
   kNonCritical,
   kDisabled,
 };
+
+const boost::unordered_map<std::string, Status> StringToStatus= {
```
Comment (alessandrothea) [inline core/include/swatch/core/AbstractMetric.hpp]: maybe better to use a std::map. boost::unordered_maps are efficient when the number of lookups is large.
Also, will this initialisation syntax work when compiled w/o c++0x flag?

File: core/include/swatch/core/MonitoringSetting.hpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "swatch/core/AbstractMetric.hpp"
+
+class MonitoringSetting {
+public:
+  MonitoringSetting(const std::string aId, const swatch::core::monitoring::Status aMonStatus);
+
+//      boost::shared_ptr<AbstractMetricCondition>,
+//      boost::shared_ptr<AbstractMetricCondition>,
+//      );
+    swatch::core::monitoring::Status getMonitoringStatus() const;
```
Comment (alessandrothea) [inline core/include/swatch/core/MonitoringSetting.hpp]: Maybe just `getStatus`

File: core/include/swatch/core/GateKeeper.hpp [hunk 3]
```diff
@@ -44,6 +44,12 @@ namespace swatch
         typedef boost::shared_ptr<tParameters> tTable;
         typedef boost::unordered_map< std::string, tTable > tTableCache;
 
+         // for monitoring settings
+        typedef tMonitoringSettingPtr tMonitoringSetting;
```
Comment (alessandrothea) [inline core/include/swatch/core/GateKeeper.hpp]: Did I alredy say tSomething is really confusing?:P

File: core/include/swatch/core/XmlGateKeeper.hpp [hunk 4]
```diff
@@ -15,35 +15,47 @@
...
 
-    GateKeeper::tTable getTable( const std::string& aKey , const std::string& aId );
+//    GateKeeper::tTable getTable( const std::string& aKey , const std::string& aId );
 
-  private:
+private:
 
-    std::pair< std::string , GateKeeper::tParameter > CreateParameter( pugi::xml_node& aEntry );
-    std::pair< std::string , GateKeeper::tTable > CreateTable( pugi::xml_node& aTable );
+  std::pair<std::string, GateKeeper::tParameter> CreateParameter(
```
Comment (alessandrothea) [inline core/include/swatch/core/XmlGateKeeper.hpp]: Why was this declaration split in 2 lines?

File: core/include/swatch/core/AbstractMetric.hpp (hunk 1 above)
Comment (kreczko) [inline core/include/swatch/core/AbstractMetric.hpp]: no :(

File: core/include/swatch/core/MonitoringSetting.hpp (hunk 2 above)
Comment (kreczko) [inline core/include/swatch/core/MonitoringSetting.hpp]: OK

File: core/include/swatch/core/GateKeeper.hpp (hunk 3 above)
Comment (kreczko) [inline core/include/swatch/core/GateKeeper.hpp]: Well, we need to agree on something :)

File: core/include/swatch/core/XmlGateKeeper.hpp (hunk 4 above)
Comment (kreczko) [inline core/include/swatch/core/XmlGateKeeper.hpp]: This was due to the previous setting to 80 chars per line

File: core/include/swatch/core/AbstractMetric.hpp (hunk 1 above)
Comment (alessandrothea) [inline core/include/swatch/core/AbstractMetric.hpp]: Ok. Can the initialisation be moved in cpp? There is no problem to initialise the map there.

Comment (kreczko) [inline core/include/swatch/core/AbstractMetric.hpp]: Does it need to? Made it static now:
http://stackoverflow.com/questions/4806510/const-variables-in-header-file-and-static-initialization-fiasco

**Pull request closed without merging**